databroker/src/grpc/kuksa_val_v2/conversions.rs

```diff
@@ -215,94 +217,170 @@ impl From<&proto::Datapoint> for broker::DataValue {
 
 impl From<&broker::Metadata> for proto::Metadata {
     fn from(metadata: &broker::Metadata) -> Self {
-        proto::Metadata {
+        let mut proto_metadata = proto::Metadata {
             id: metadata.id,
             data_type: proto::DataType::from(metadata.data_type.clone()) as i32,
             entry_type: proto::EntryType::from(metadata.entry_type.clone()) as i32,
             description: Some(metadata.description.clone()),
             comment: None,
             deprecation: None,
             unit: metadata.unit.clone(),
-            value_restriction: match metadata.allowed.as_ref() {
-                Some(allowed) => match allowed {
-                    broker::DataValue::StringArray(vec) => Some(proto::ValueRestriction {
+
+            // We add value_restriction below
+            value_restriction: None,
```

Comment:
I think it might be a bit easier to follow if you moved the conversion to a `value_restriction` to a separate function and just assigned it here.